### PR TITLE
[QWen3VL] Adding CUDA Graph Support for LLM Sub-module and Vision Encoder Sub-module

### PR DESCRIPTION
**Pull request by tomlifu**
# What does this PR do ?

This PR adds CUDA Graph support for vision encoder. 
Previous PR: https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/2274 was deprecated.

Related Mcore PR:https://github.com/NVIDIA/Megatron-LM/pull/3293, https://github.com/NVIDIA/Megatron-LM/pull/3294
Related TE PR:https://github.com/NVIDIA/TransformerEngine/pull/2657

Example run command:

```
uv run python -m torch.distributed.run --nproc_per_node=8 ${MEGATRON_BRIDGE_PATH}/scripts/training/run_recipe.py \
  --recipe qwen3_vl_8b_pretrain_mock_config \
  --step_func qwen3_vl_step \
  dataset.seq_length=8192 \
  dataset.num_images=1 \
  dataset.image_size=[768,768] \
  dataset.dataloader_type="cyclic" \
  optimizer.clip_grad=0 \
  model.seq_length=8192 \
  model.num_layers=8 \
  model.freeze_language_model=False \
  model.freeze_vision_model=False \
  model.freeze_vision_projection=False \
  model.pipeline_model_parallel_size=2 \
  model.tensor_model_parallel_size=1 \
  model.expert_model_parallel_size=1 \
  model.cuda_graph_impl="transformer_engine" \
  model.cuda_graph_scope=[attn,mlp] \
  model.vision_cuda_graph_impl=transformer_engine \
  model.vision_cuda_graph_scope=[attn,mlp] \
  model.max_vision_cuda_graph_seq_length=4608 \
  model.use_te_rng_tracker=True \
  model.apply_rope_fusion=False \
  train.micro_batch_size=2 \
  train.global_batch_size=32 \
  train.train_iters=20 \
  train.eval_iters=0 \
  train.manual_gc=true \
  train.manual_gc_interval=100 \
```

# Changelog

- Add specific line by line info of high level changes in this PR.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

## Summary by CodeRabbit

* **New Features**
  * Added vision encoder CUDA graph support for improved training performance.
  * Added configurable vision CUDA graph settings including implementation scope and maximum sequence length limits.

* **Improvements**
  * Enhanced vision model with automatic sequence padding for CUDA graph compatibility.

* **Updates**
  * Updated Megatron-LM submodule to latest version.

### CODE REVIEW

**Comment by shifangx** [comment]
We need make sure the lm loss remain the same after using cuda graph.

File: src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/vision_model.py [hunk 1]
```diff
@@ -235,17 +250,75 @@ def forward(
...
+                    (1, 1, seq_len, seq_len),
+                    dtype=hidden_states.dtype,
+                    device=hidden_states.device
+                )
+                # Mask out padding columns (keys)
+                attention_mask[:, :, :, original_seq_len:] = 0
+                # Mask out padding rows (queries) - these will produce garbage anyway
+                attention_mask[:, :, original_seq_len:, :] = 0
+                # Convert to additive mask (0 -> 0, 1 -> -inf for softmax)
+                attention_mask = (1.0 - attention_mask) * torch.finfo(hidden_states.dtype).min
```
Comment (shifangx) [inline src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/vision_model.py]: Can we use this attention_mask for both of the cuda graph case and the non cuda graph case?
So we can delete funtion build_packed_seq_params.

CC @wplf

**Comment by tomlifu** [comment]
> We need make sure the lm loss remain the same after using cuda graph.

I have made a fix in my mcore PR: https://github.com/NVIDIA/Megatron-LM/pull/3294

File: src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/vision_model.py (hunk 1 above)
Comment (tomlifu) [reply src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/vision_model.py]: I don't think we can completely get rid of build_packed_seq_params. If we have multiple images, we need to build packed_sequence to process them.

File: src/megatron/bridge/training/train.py [hunk 2]
```diff
@@ -250,6 +258,36 @@ def train(
...
+                            model=model,
+                            vision_config=vision_model_config,
+                            vision_seq_length=vision_seq_length,
+                            micro_batch_size=config.train.micro_batch_size,
+                            num_microbatches=get_num_microbatches(),
+                        )
+                        print_rank_0(
+                            f"Vision encoder CUDA graph enabled with seq_length={vision_seq_length}"
+                        )
+                    break
```
[the remark this answers is not in the document]
Comment (shifangx) [reply src/megatron/bridge/training/train.py]: get_attr_wrapped_model(..., "vision_model") returns the object that has vision model as an attribute, not returns vision model itself.

**Review by yaoyu-33 — APPROVED** [review]
